### ISSUE TITLE
Cannot create Route53 automatically DNS-validated certificate

### ISSUE THREAD

**Issue by realharry** [aws/aws-cdk#3592]
**Note: for support questions, please first reference our [documentation](https://docs.aws.amazon.com/cdk/api/latest), then use [Stackoverflow](https://stackoverflow.com/questions/ask?tags=aws-cdk)**. This repository's issues are intended for feature requests and bug reports.

* **I'm submitting a ...**
  - [x] :beetle: bug report
  - [ ] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*

Create a public hosted zone, or look up a `zone` using `fromLookup()` (e.g., for "mydomain.com".)

Try to create a certificate using the hosted zone, e.g.,
```BASH
new certmgr.DnsValidatedCertificate(this, certId, {
      domainName: 'testing.mydomain.com',
      hostedZone: zone,
```

`cdk deploy` throws error:

```BASH
 5/7 | 4:29:29 PM | CREATE_IN_PROGRESS   | AWS::CloudFormation::CustomResource | stagingCertificate/CertificateRequestorResource/Default (stagingCertificateCertificateRequestorResourceE148D233)
 5/7 | 4:29:34 PM | CREATE_IN_PROGRESS   | AWS::CloudFormation::CustomResource | stagingCertificate/CertificateRequestorResource/Default (stagingCertificateCertificateRequestorResourceE148D233) Resource creation Initiated
 6/7 | 4:29:35 PM | CREATE_FAILED        | AWS::CloudFormation::CustomResource | stagingCertificate/CertificateRequestorResource/Default (stagingCertificateCertificateRequestorResourceE148D233) Failed to create resource. Cannot read property '0' of undefined
	new CustomResource (/Users/harry/Projects/devops/deploy/rbi0-acm/node_modules/@aws-cdk/aws-cloudformation/lib/custom-resource.ts:92:21)
	\_ new DnsValidatedCertificate (/Users/harry/Projects/devops/deploy/rbi0-acm/node_modules/@aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts:73:29)
	\_ Rbi0AcmStack.createCertificate (/Users/harry/Projects/devops/deploy/rbi0-acm/lib/rbi0-acm-stack.ts:46:25)
	\_ Rbi0AcmStack.init (/Users/harry/Projects/devops/deploy/rbi0-acm/lib/rbi0-acm-stack.ts:34:30)
	\_ new Rbi0AcmStack (/Users/harry/Projects/devops/deploy/rbi0-acm/lib/rbi0-acm-stack.ts:24:10)
	\_ Object.<anonymous> (/Users/harry/Projects/devops/deploy/rbi0-acm/bin/rbi0-acm.ts:13:1)
	\_ Module._compile (internal/modules/cjs/loader.js:701:30)
	\_ Module.m._compile (/Users/harry/Projects/devops/deploy/rbi0-acm/node_modules/ts-node/src/index.ts:473:23)
	\_ Module._extensions..js (internal/modules/cjs/loader.js:712:10)
	\_ Object.require.extensions.(anonymous function) [as .ts] (/Users/harry/Projects/devops/deploy/rbi0-acm/node_modules/ts-node/src/index.ts:476:12)
	\_ Module.load (internal/modules/cjs/loader.js:600:32)
	\_ tryModuleLoad (internal/modules/cjs/loader.js:539:12)
	\_ Function.Module._load (internal/modules/cjs/loader.js:531:3)
	\_ Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
	\_ Object.<anonymous> (/Users/harry/Projects/devops/deploy/rbi0-acm/node_modules/ts-node/src/bin.ts:158:12)
	\_ Module._compile (internal/modules/cjs/loader.js:701:30)
	\_ Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
	\_ Module.load (internal/modules/cjs/loader.js:600:32)
	\_ tryModuleLoad (internal/modules/cjs/loader.js:539:12)
	\_ Function.Module._load (internal/modules/cjs/loader.js:531:3)
	\_ Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
	\_ findNodeScript.then.existing (/usr/local/lib/node_modules/npm/node_modules/libnpx/index.js:268:14)
```

* **What is the expected behavior (or behavior of feature suggested)?**

According to to the doc, https://docs.aws.amazon.com/cdk/api/latest/docs/aws-certificatemanager-readme.html, you should be able to create a certificate based on an existing public hosted zone. Or, at least, we should be notified (I presume) for DNS validation.

* **What is the motivation / use case for changing the behavior or adding this feature?**

Need to env-specific certificates during deploy.

* **Please tell us about your environment:**

  - CDK CLI Version: 1.3.0 (build bba9914)
  - Module Version: 1.3.0
  - OS: OSX Mojave
  - Language: TypeScript

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

**Comment by jordankueh** [comment]
I might be loopy (@samjarrett claims I definitely am), but I had this working at one point a few weeks ago.

In the last week, it's ceased to function, and I'm getting the exact same error (`Cannot read property '0' of undefined`).

Looking through the template, the only index references I can see are for the `CertificateRequestorFunction`, in order to reference the correct bucket path. The parameter it's referencing is supplied and visible in the CloudFormation console.

... Which makes me think it's part of the Lambda code itself - Looking through it after downloading it from the bucket:

![image](https://user-images.githubusercontent.com/1776706/62751407-f3da0580-baa6-11e9-8d37-11b7bebb1607.png)

There appears to be this little snippet on lines 107-110 (for v1.3.0) inclusive that make reference to the `DomainValidationOptions` for the certificate.

My running theory is that the `DomainValidationOptions` for the certificate is not available when this runs as ACM has not yet finished creating the validation records, and attempts the check too early, and so fails at this step.

**Comment by nsvarich** [comment]
I started running into this same issue (a previously working CDK deploy suddenly stopped working with this error) and discovered the cause to be my new ACM certificates were stuck in status "Pending Validation", due to I believe hitting a limit of Route 53 record sets. After deleting several old CNAME records this issue went away.

It is possible you're receiving the `Failed to create resource. Cannot read property '0' of undefined` error from a different cause, but this appears to be one of the possible causes of this error.

**Comment by eladb** [comment]
I believe this is fixed by #3528. Closing for now. Reopen as needed.
